lib/sisimai/bite/email/zoho.rb

```diff
@@ -0,0 +1,167 @@
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        qprintable = false
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # This message was created automatically by mail delivery software.
+            # A message that you sent could not be delivered to one or more of its recip=
+            # ients. This is a permanent error.=20
+            #
+            # kijitora@example.co.jp Invalid Address, ERROR_CODE :550, ERROR_CODE :5.1.=
+            # 1 <kijitora@example.co.jp>... User Unknown
+
+            # This message was created automatically by mail delivery software.
+            # A message that you sent could not be delivered to one or more of its recipients. This is a permanent error.
+            #
+            # shironeko@example.org Invalid Address, ERROR_CODE :550, ERROR_CODE :Requested action not taken: mailbox unavailable
+            v = dscontents[-1]
+
+            if cv = e.match(/\A([^ ]+[@][^ ]+)[ \t]+(.+)\z/)
+              # kijitora@example.co.jp Invalid Address, ERROR_CODE :550, ERROR_CODE :5.1.=
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              v['diagnosis'] = cv[2]
+
+              if v['diagnosis'] =~ /=\z/
+                # Quoted printable
+                v['diagnosis'] = v['diagnosis'].sub(/=\z/, '')
+                qprintable = true
+              end
+              recipients += 1
+
+            elsif cv = e.match(/\A\[Status: .+[<]([^ ]+[@][^ ]+)[>],/)
+              # Expired
+              # [Status: Error, Address: <kijitora@6kaku.example.co.jp>, ResponseCode 421, , Host not reachable.]
+              if v['recipient']
```

Comment:
__[rubocop]__
* Performance/EndWith: Use `String#end_with?` instead of a regex match anchored to the end of the string. [:ref](https://github.com/JuanitoFatas/fast-ruby#stringmatch-vs-stringstart_withstringend_with-code-start-code-end)

<!-- Comment by [SideCI](https://sideci.com) -->